platform/windows/display_server_windows.cpp

```diff
@@ -546,6 +546,17 @@ DisplayServer::WindowID DisplayServerWindows::create_sub_window(WindowMode p_mod
 		wd.no_focus = true;
 	}
 
+	// Inherit icons from MAIN_WINDOW for all sub windows that are not borderless
+	if (!wd.borderless) {
```

Comment:
Are there any issues with setting the icon of a borderless window? If there are, `window_set_flag` should be also modified to set/unset icon when borderless mode is toggled. If not (which is probably the case) just remove this check.